application-workloads/azure-gamedev/gamedev-vm/nestedtemplates/gamedev-vm.bicep

```diff
@@ -0,0 +1,412 @@
+    }
+  }
+  production: {
+    vmImage: {
+      publisher: 'microsoftcorporation1602274591143'
+      offer: 'game-dev-vm'
+      sku: '${osType}_${environmentMapping[gameEngine]}'
+      version: 'latest'
+    }
+    vmPlan: {
+      publisher: 'microsoftcorporation1602274591143'
+      product: 'game-dev-vm'
+      name: '${osType}_${environmentMapping[gameEngine]}'
+    }
+  }
+}
+
+var vmImage = environments[environment].vmImage
+var vmPlan = environments[environment].vmPlan
+
+var ipconfName = '${vmName}-ipconf'
+var nicName = '${vmName}-nic'
+var nsgName = '${vmName}-nsg'
+
+var storageType = (bool(length(split(vmSize, '_')) > 2) ? 'Premium_LRS' : 'Standard_LRS')
+
+var tags = {
+  'solution': 'Game Development Virtual Machine'
+  'engine': gameEngine
+  'ostype': osType
+  'remotesoftware': remoteAccessTechnology
+}
+
+var cmdGDKInstall = '(Get-Content \'C:\\ProgramData\\Microsoft\\Windows\\Start Menu\\Programs\\Startup\\gdkinstall.cmd\').replace(\'[VERSION]\', \'${gdkVersion}\') | Set-Content \'C:\\ProgramData\\Microsoft\\Windows\\Start Menu\\Programs\\Startup\\gdkinstall.cmd\''
+var userData = 'team_id=${parsec_teamId}:key=${parsec_teamKey}:name=${parsec_host}:user_email=${parsec_userEmail}:is_guest_access=${parsec_isGuestAccess}:ibLicenseKey=${ibLicenseKey}'
+var Script2Run = 'TeradiciRegCAS.ps1'
+var CSEParams = ' -pcoip_registration_code ${teradiciRegKey}'
+var cmdTeradiciRegistration = './${Script2Run}${CSEParams}'
+
+var vnetId = {
+  'new'     : resourceId('Microsoft.Network/virtualNetworks', vnetName)
+  'existing': resourceId(vnetRGName, 'Microsoft.Network/virtualNetworks', vnetName)
+}
+var subnetId = '${vnetId[vnetNewOrExisting]}/subnets/${subNetName}'
+
+var publicIpId = {
+  'new': resourceId('Microsoft.Network/publicIPAddresses', publicIpName)
+  'existing': resourceId(publicIpRGName, 'Microsoft.Network/publicIPAddresses', publicIpName) 
+  'none': ''
+}[publicIpNewOrExisting]
+
+resource partnercenter 'Microsoft.Resources/deployments@2021-04-01' = {
+  name: 'pid-7837dd60-4ba8-419a-a26f-237bbe170773-partnercenter'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+    }
+  }
+}
+
+resource publicIp 'Microsoft.Network/publicIPAddresses@2021-05-01' = if (publicIpNewOrExisting == 'new') {
+  name: publicIpName
+  location: location
+  sku: {
+    name: publicIpSku
+  }
+  properties: {
+    publicIPAllocationMethod: publicIpAllocationMethod
+    dnsSettings: {
+      domainNameLabel: publicIpDns
+    }
+  }
+}
+
+module nsg_rules 'nsgRules.bicep' = {
+  name: 'nsg_rules'
+  params: {
+    addPixelStreamingPorts: unrealPixelStreamingEnabled
+  }
+}
+
+resource nsg 'Microsoft.Network/networkSecurityGroups@2021-05-01' = {
+  name: nsgName
+  location: location
+  properties: {
+    securityRules: nsg_rules.outputs.nsgRules['nsgRules-${remoteAccessTechnology}']
```

Comment:
this is more indirect than we should be - it creates a short circuit of the deployment validation (because this module relies on an input from another module...  just move the logic here.

Reply:
- deleted nsg_rules.bicep
- moved var from nsg_rules.bicep to gamedev-vm.bicep